pkg/bootstrap/platform/gcp.go

```diff
@@ -376,7 +378,14 @@ func createMetadataSupplier(property string, fn func() (string, error)) metadata
 }
 
 func isMetadataEndpointAccessible() bool {
-	_, err := net.DialTimeout("tcp", "metadata.google.internal:80", 5*time.Second)
+	// From the Go package, but private so copied here
+	const metadataHostEnv = "GCE_METADATA_HOST"
+	const metadataIP = "169.254.169.254"
+	host := os.Getenv(metadataHostEnv)
+	if host == "" {
+		host = metadataIP
```

Comment:
As mentioned in chat, I think we need to check the additional condition that the header of the returned result matches the GCP metadata flavor header.

Reply:
I think if we do that we need to completely replace OnGCE. I might not have time to cover that in the short term future, does it make sense as a followup? Since right now tests are broken. The current change _should_ be strictly an improvement